### DISCUSSION TITLE
Angular-cli installs node even tho node is installed through n

### DISCUSSION THREAD

**Discussion by ScriptType** [Homebrew/discussions#5704] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: 91f3671b16fe62121364c465c14e0dfd088d7ead
Last commit: 4 days ago
Core tap JSON: 31 Oct 20:16 UTC
Core cask tap JSON: 31 Oct 20:16 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: code-insiders
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.3.5 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 16.0.0 build 1600
Git: 2.47.0 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.0.1-arm64
CLT: 16.0.0.0.1.1724870825
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1
```

### Description of issue

I installed n through brew and want to manage with this my node environment.
But when I install angular afterwards it always installs also node (also node 23 which is not recommended for production.)
Sure, I could install angular with NPM, but I like to do brew upgrade to update my angular-cli etc.

Is there some solution for this?

**Comment by gromgit** [comment]
I don't think so. All Node-dependent formulae rely on Homebrew-built Nodes for reliability and reproducibility reasons (any Homebrew Node upgrade will trigger a retest against all its dependencies), so all such formulae will have a stable reference Node that can be counted on, _and which can conceivably differ between formulae_.

`n` allows users to swap out versions at will, nullifying that guarantee and potentially causing formulae to break without warning.

_Sidenote:_ I'm not an active Node user, but I absolutely disagree with the recommended solution in [Switching to `n` Managed Node.js](https://github.com/tj/n/blob/master/docs/changing-node-location.md). Disk space is cheap, time spent debugging sudden failures when you activate a new Node version that hasn't been tested against third-party software is bloody expensive. Far better to let Homebrew manage the Node installs that Homebrew formulae have been built and tested against, let `n` manage the Node(s) that _you_ personally want, and swallow the extra space requirements.
